File: C_Sharp/Framework.Tests/CustomerServices/GetCustomerByIdQuery.cs
using System;
using System.Collections.Generic;
using Framework.Data_Manipulation;

namespace Framework.Tests.CustomerServices
{
    class GetCustomerByIdQuery : BaseQueryObject<Customer, GetCustomerByIdQuery.Criteria>
    {
        public class Criteria
        {
            public int CustomerId { get; set; }

            private Criteria()
            {
            }

            public static Criteria SearchById(int id)
            {
                return new Criteria {CustomerId = id};
            }
        }

        public override IList<Customer> PerformSearchOperation(Criteria searchInput)
        {
            Dictionary<int, string> customerList = new Dictionary<int, string>();

            customerList.Add(1, "Juan dela Cruz");
            customerList.Add(2, "Jane Doe");
            customerList.Add(3, "John Doe");

            Tuple<string, string> searchResult = new Tuple<string, string>(searchInput.CustomerId.ToString(), customerList[searchInput.CustomerId]);
            IBaseMapper mapper = DataSynchronizationManager.GetInstance().GetMapper<Customer>();
            Customer customer = new Customer(mapper);

            customer.Number = searchResult.Item1;
            customer.Name = searchResult.Item2;

            return new List<Customer>(new[] { customer });
        }
    }
}

File: Android/app/src/test/java/com/CustomerServices/GetCustomerByIdQuery.java
package com.CustomerServices;

import com.DataManipulation.BaseMapperInterfaces.IBaseMapper;
import com.DataManipulation.BaseQueryObject;
import com.DataSynchronizationManager;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

/**
 * Created by aiko on 4/8/17.
 */

public class GetCustomerByIdQuery extends BaseQueryObject<Customer, GetCustomerByIdQuery.Criteria> {

    public static class Criteria {
        public int CustomerId;

        public Criteria(int customerId) {
            CustomerId = customerId;
        }
    }

    public GetCustomerByIdQuery() {
        super( GetCustomerByIdQuery.Criteria.class);
    }

    @Override
    public List<Customer> PerformSearchOperation(GetCustomerByIdQuery.Criteria criteria) {
        HashMap<Object, String> customerList = new HashMap<Object, String>();


        customerList.put(1, "Juan dela Cruz");
        customerList.put(2, "Jane Doe");
        customerList.put(3, "John Doe");

        String searchResult = customerList.get(criteria.CustomerId);
        IBaseMapper mapper = DataSynchronizationManager.GetInstance().GetMapper(Customer.class);
        Customer customer = new Customer(mapper);

        customer.Number = String.valueOf(criteria.CustomerId);
        customer.Name = searchResult;

        List<Customer> customerResult = new ArrayList<Customer>();

        customerResult.add(customer);

        return customerResult;
    }
}
